app/src/main/res/layout/current_weather.xml

```diff
@@ -18,11 +18,11 @@
         android:layout_height="wrap_content"
         android:layout_marginStart="16dp"
         android:layout_marginTop="36dp"
-        android:text='@{Double.toString(mainScreen.weatherData.main.temp)+` C`}'
+        android:text='@{Double.toString(viewModel.weatherData.main.temp)+` C`}'
```

Comment:
We should simplify using calculations inside xml. The best way is to have LiveData for every field. Use one base MutableLiveData and Transformations for other fields

LiveData<User> userLiveData = ...;
LiveData<String> userFullNameLiveData =
    Transformations.map(
        userLiveData,
        user -> user.firstName + user.lastName);
});

https://developer.android.com/reference/androidx/lifecycle/Transformations